apps/src/lib/node/ledger/shell/finalize_block.rs

```diff
@@ -339,8 +340,16 @@ where
                 },
             };
 
-            match protocol::apply_tx(tx_type, tx_length, self.into())
-                .map_err(Error::TxApply)
+            match protocol::dispatch_tx(
+                tx_type,
+                tx_length,
+                &mut self.gas_meter,
+                &mut self.write_log,
+                &mut self.storage,
+                &mut self.vp_wasm_cache,
+                &mut self.tx_wasm_cache,
+            )
+            .map_err(Error::TxApply)
```

Comment:
without `ShellParams` this call becomes huge. well it's whatever really

Reply:
yeah we added `ShellParams` once we needed the extra `wasm_dir` var (due to a clippy warning I think)? This code now is more similar to how it is in `main`, using this long list of parameters as well. I think we could later on further refactor `dispatch_tx` to be a method on `Shell` or something like that since it uses so many of its fields.

Reply:
I like the shell params. It it tightens the code up a bit.

Reply:
I put back `ShellParams` though for `dispatch_tx` we still need a `&mut Storage` whereas `ShellParams` has an immutable `&Storage`. So here is still requiring to pass the params explicitly but at the other two call sites where we're changing to calling `apply_wasm_tx`, those can stay the same.